zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -305,6 +320,9 @@ fn init_wallet_db_internal<P: consensus::Parameters + 'static>(
 ) -> Result<(), MigratorError<WalletMigrationError>> {
     let seed = seed.map(Rc::new);
 
+    // Verify that the sqlite version in use supports the features required by this library
+    verify_sqlite_version_compatibility(&wdb.conn).map_err(MigratorError::Adapter)?;
```

Comment:
Since we've established that the issue was with the SQLite version used by the app to run queries, *not* the version used by `rusqlite`, is this even necessary any more? This query will always check only the version of the SQLite used by `rusqlite` (which is the bundled one, since we enable the `bundled` feature).

We know that the [required](https://github.com/zcash/librustzcash/blob/50cdf73dfdd13828995b06a7989c8c0fa10c27d4/Cargo.toml#L125) and pinned `rusqlite` (version 0.29.0) [bundles SQLite version 3.41.2](https://crates.io/crates/rusqlite/0.29.0#notes-on-building-rusqlite-and-libsqlite3-sys), so that is the version we are guaranteed to get currently. (Technically a `rusqlite` version 0.29.x could be released that bundles a different version and would be picked up by a `cargo update`, or we could increase the rusqlite version, but those would only be likely to bundle a newer SQLite.)

Reply:
I added this essentially as a line of defense. I don't think that this query of constant data is an unreasonable guardrail, particularly given that it's only executed on wallet startup (and ideally would only be executed on wallet installation or upgrade).

Reply:
Nod. In that case please log the version and add a comment about it being the version used by `rusqlite`. (Ideally I would also like to log the version used by the app, but that's an SDK issue.)

Reply:
We should also document what SQLite version we're relying on for the view queries, and then go through those checking that we're not relying on newer features. Is this PR expected to make it work on API level 27 (which provides at least SQLite 3.19.4 according to https://stackoverflow.com/a/4377116/393146 )?